tests/test_device.py

```diff
@@ -89,6 +89,8 @@ def test_query_gid_table(self):
         with d.Context(name=devs[0].name.decode()) as ctx:
             device_attr = ctx.query_device()
             port_attr = ctx.query_port(1)
+            if port_attr.link_layer == e.IBV_LINK_LAYER_UNSPECIFIED:
```

Comment:
Why can you assume that ibv_query_gid_table is not supported on IBV_LINK_LAYER_UNSPECIFIED? AFAIK, this type of link layer is a legacy value used to indicate that the link layer is IB/iWARP

Reply:
I assume legacy code would return IBV_GID_TYPE_SYSFS_IB_ROCE_V1, which returns EINVAL in case of IBV_LINK_LAYER_UNSPECIFIED (in query_sysfs_gid_entry()).